app/questionnaire/path_finder.py

```diff
@@ -164,36 +195,38 @@ def _remove_rule_answers(self, goto_rule, this_location):
                 if 'meta' not in condition.keys():
                     self.answer_store.remove_answer(condition['id'])
 
-        if this_location in self.completed_blocks:
-            self.completed_blocks.remove(this_location)
+        if this_location in self.completed.blocks:
+            self.completed.blocks.remove(this_location)
 
-    def get_full_routing_path(self):
+    def routing_path(self, current_section: Mapping) -> RoutingPath:
         """
         Returns a list of the block ids visited based on answers provided
         :return: List of block location dicts
         """
-        if self._full_routing_path and not self.answer_store.is_dirty:
-            return self._full_routing_path
+        if self._routing_path and not self.answer_store.is_dirty:
```

Comment:
I wonder if `self._routing_path` should be a map? keyed by the section?
that way we can potentially cache the path for multiple sections.

Reply:
It probably should. It would be useful to have the full path followed by the user in a number of scenarios.